code/plugin/src/main/kotlin/com/android/gradle/replicator/codegen/plugin/CodegenPlugin.kt

```diff
@@ -0,0 +1,159 @@
+                t.componentFilter { componentIdentifier ->
+                    when (componentIdentifier) {
+                        is ProjectComponentIdentifier -> apiModules.contains(componentIdentifier.projectPath)
+                        else -> false
+                    }
+                }
+                t.attributes(attributeFilter)
+            }.artifacts
+
+            val implProjectArtifacts = config.incoming.artifactView { t: ArtifactView.ViewConfiguration ->
+                t.componentFilter { componentIdentifier ->
+                    when (componentIdentifier) {
+                        is ProjectComponentIdentifier -> !apiModules.contains(componentIdentifier.projectPath)
+                        else -> false
+                    }
+                }
+                t.attributes(attributeFilter)
+            }.artifacts
+
+            // create an artifact collection for both Api and Implementation.
+            val apiArtifacts = config.incoming.artifactView { t: ArtifactView.ViewConfiguration ->
+                t.componentFilter(apiFilter)
+                t.attributes(attributeFilter)
+            }.artifacts
+
+            val implArtifacts = config.incoming.artifactView { t: ArtifactView.ViewConfiguration ->
+                t.componentFilter(implFilter)
+                t.attributes(attributeFilter)
+            }.artifacts
+
+            // get the android APIs, this will be added to our classloader and API which is done automatically by AGP.
+            val androidApis = project.configurations.getAt("androidApis").incoming.files
+
+            config.allDependencies.forEach {
+                task.gradleDependencies.add("${it.group}:${it.name}:${it.version}")
+            }
+
+
+            val runtimeClasspath = project.configurations.getAt("debugRuntimeClasspath")
+            val runtimeClasspathArtifacts = runtimeClasspath.incoming.artifactView {
+                it.attributes(attributeFilter)
+            }.artifacts
+
+            task.runtimeClasspath.from(runtimeClasspathArtifacts.artifactFiles)
+            task.codeGeneratedModuleApiClasspath.from(apiProjectArtifacts.artifactFiles)
+            task.codeGeneratedModuleImplClasspath.from(implProjectArtifacts.artifactFiles)
+            task.apiJarFiles.from(apiArtifacts.artifactFiles)
+            task.implJarFiles.from(androidApis, implArtifacts.artifactFiles)
+            task.paramsFile.set(project.layout.buildDirectory.file("test.params"))
+            // Randomizer values should be set during project replication along the number of java and kotlin files.
+            task.seed.set(Random.nextInt())
+            // make sure we depend on our dependencies built artifacts so we have access to their generated classes.
+            projectDependencies.forEach {
+                // hack : use Variant API when dealing with android.
+                task.dependsOn(if (project.tasks.findByName("$it:assembleDebug") != null) {
```

Comment:
also, this force creation of the task. Sadly there's not a better API at the moment.

Reply:
I should be able to depend on AAR, i'll look into it.